metals/src/main/scala/scala/meta/internal/parsing/JavaFoldingRangeExtractor.scala

```diff
@@ -46,14 +46,16 @@ final class JavaFoldingRangeExtractor(
   def extract(): List[FoldingRange] = {
     val scanner = ToolFactory.createScanner(true, true, false, true)
     scanner.setSource(text.toCharArray())
-
     @tailrec
     def swallowUntilSemicolon(token: Int, line: Int): Int = {
-      val addLine = scanner.getCurrentTokenSource.count(_ == '\n')
-      if (token != ITerminalSymbols.TokenNameSEMICOLON) {
-        swallowUntilSemicolon(scanner.getNextToken(), line + addLine)
-      } else {
-        line + addLine
+      if (token == ITerminalSymbols.TokenNameEOF) line
```

Comment:
I was looking at errors in `metals.log` to decide where to create reports and this was throwing an `IndexOutOfBounds` for `scanner.getCurrentTokenSource`.

Reply:
Good catch!